packages/core/helper-plugin/lib/src/components/GenericInput/index.js

```diff
@@ -216,12 +216,12 @@ const GenericInput = ({
           hint={hint}
           name={name}
           onValueChange={value => {
-            onChange({ target: { name, value, type } });
+            onChange({ target: { name, value: value ?? null, type } });
           }}
           placeholder={formattedPlaceholder}
           required={required}
           step={step}
-          value={value ?? undefined}
+          value={value ?? null}
```

Comment:
This will never be applied.

If you check out the defaultProps you'll see that the `value` is set to `''` when it is not passed to the GenericInput so when it is `undefined` .

So the `??` will not work as the value will always be an empty string.

Maybe you could update the defaultProps.value to set it to `undefined` ?

Reply:
It wasn't that easy actually. In 004ce7ea72673dcba23acf873456e7ff795ff054 I made an attempt to unify the handling a bit for the whole file:

- the API sends always `null` in case the value is empty, which is not allowed in React
- therefore every nullish value is now casted to `undefined`
- in case a value needs to be reset, the value has to be `null`, otherwise some components stopped working

I think this is easier to understand and less individual checking. What do you think?